### ISSUE TITLE
Client server delegation base URL is not an URL

### ISSUE THREAD

**Issue by ChrisLahaye** [matrix-org/dendrite#2941]
### Background information

- **Dendrite version or git SHA**: 0.10.8
- **Monolith or Polylith?**: Monolith
- **SQLite3 or Postgres?**: Postgres
- **Running in Docker?**: Yes

### Description

- **What**:  The base URL returned by /.well-known/matrix/client is not an URL and misses a schema, instead, it just returns the well known client name. 
- **Who**: Clients
- **How**: Can't login on Element on a server with client-server delegation

### Steps to reproduce

**Minimal**: 
- set well known client name to matrix-client.example.com:443
- inspect /.well-known/matrix/client
- observe that base URL is not an URL

**Full**: 
- set server name to example.com
- set well known server name to matrix-server.example.com:443
- set well known client name to matrix-client.example.com:443
- attempt to login on the server using Element

See https://github.com/matrix-org/dendrite/blob/f762ce1050f2add409a83b1eeb6da5940177cfa7/clientapi/routing/routing.go#L100

**Comment by ChrisLahaye** [comment]
I can create a pull request for this, just need a confirmation and some help on how to determine the proper schema. Is this always https:// and can it be hard-coded, or is there a way to determine the right schema? If not, then potentially we could refactor well known client name into well known client base url?

**Comment by bones-was-here** [comment]
The config documentation is not consistent with the behaviour.  People do have this working currently, eg:
```
  well_known_server_name: "example.com:443"     # {"m.server":"example.com:443"}
  well_known_client_name: "https://example.com" # {"m.homeserver":{"base_url":"https://example.com"}}
```

**Comment by ChrisLahaye** [comment]
@bones-was-here thank you for your reply. That is indeed how we are working around the issue as well.

**Comment by S7evinK** [comment]
Documentation is going to be fixed with #2953, we can also verify that `cfg.Matrix.WellKnownClientName` is in fact a URL on startup.
